src/dirwatcher/index.ts

```diff
@@ -0,0 +1,70 @@
+import EventEmitter from 'events';
+import path from 'path';
+import fs from 'fs';
+import logger from '../../logger';
+import { DIRECTORY_FILES_CHANGED } from 'constants/EventNames';
+
+export default class DirWatcher extends EventEmitter {
+    private changedFiles = [];
+    private allFilesInDirectoryStats = {};
+
+    public watch(dirPath: string, delay: number) {
+        setInterval(() => this.readDirectory(dirPath), delay);
+    }
+
+    private readDirectory(dirPath: string) {
```

Comment:
this method does a way above than it's declared in it's name

Reply:
will consider renaming